operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java

```diff
@@ -195,4 +249,14 @@ private void throwMissingCRDException(String crdName, String specVersion, String
     }
     return false;
   }
+
+  private static class ControllerRef {
```

Comment:
This should be parameterized by the CR type so that the `ResourceController` and `ControllerConfiguration` instances use the same type…

Reply:
I'm not sure how it should be done as the list of ControllerRef would need to be on the base type in any case. Unfortunately, the usage of generic in the project is a little bit confused an inconsistent so the only way I found to make all the methods happy were to remove generics. 

But I'm happy to change this if you help me understanding the right way of doing it.

Reply:
I agree that the generic usage is inconsistent in the project. Let's address this in a subsequent PR.

Reply:
thx, can you re-trigger the CI ?